File: API/Entities/OrderAggregate/Order.cs
namespace API.Entities.OrderAggregate
{
    public class Order
    {
        public int Id { get; set; }
        public string BuyerId { get; set; }
        public ShippingAddress ShippingAddress { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        public List<OrderItem> Items { get; set; } = new();
        public decimal Subtotal { get; set; }
        public decimal DeliveryFee { get; set; }
        public OrderStatus Status { get; set; } = OrderStatus.Pending;
        //public string? PaymentIntentId { get; set; }

        public decimal GetTotal()
        {
            return Subtotal + DeliveryFee;
        }

    }
}

File: client/src/app/models/order.ts
import { ShippingAddress } from "./shippingAddress";

export interface OrderItem {
    productId: number;
    name: string;
    pictureUrl: string;
    price: number;
    quantity: number;
}

export interface Order {
    id: number;
    buyerId: string;
    shippingAddress: ShippingAddress;
    orderDate: string;
    items: OrderItem[];
    subtotal: number;
    deliveryFee: number;
    status: string;
    total: number;
}

export interface OrderParams {
    orderBy: string,
    searchTerm?: string,
    status?: number,
    pageNumber: number,
    pageSize: number,
}

export enum OrderStatus {
    Pending,
    Confirmed,
    Delivered,
    Cancelled,
}

export interface OrderTotals {
    count: number;
    total: number;
}